nucypher/characters/lawful.py

```diff
@@ -1735,6 +1735,62 @@ def _reencrypt(self, kfrag: KFrag, work_order: 'WorkOrder', alice_verifying_key:
         # ... and finally returns all the re-encrypted bytes
         return cfrag_byte_stream
 
+    def status_info(self, omit_known_nodes: bool = False) -> 'LocalUrsulaStatus':
+
+        domain = self.domain
+        version = nucypher.__version__
+
+        latest_states = self.known_nodes.latest_states(5)
+
+        fleet_state = latest_states[-1]
+        previous_fleet_states = [state for state in latest_states[:-1]]
+
+        if not omit_known_nodes:
+            known_nodes_info = [self.known_nodes.status_info(node) for node in self.known_nodes]
+        else:
+            known_nodes_info = None
+
+        return LocalUrsulaStatus(nickname=self.nickname,
+                                 staker_address=self.checksum_address,
+                                 worker_address=self.worker_address,
+                                 rest_url=self.rest_url(),
+                                 timestamp=self.timestamp,
+                                 domain=domain,
+                                 version=version,
+                                 fleet_state=fleet_state,
+                                 previous_fleet_states=previous_fleet_states,
+                                 known_nodes=known_nodes_info,
+                                 )
+
+
+class LocalUrsulaStatus(NamedTuple):
+    nickname: Nickname
+    staker_address: ChecksumAddress
+    worker_address: str
+    rest_url: str
+    timestamp: maya.MayaDT
+    domain: str
+    version: str
+    fleet_state: ArchivedFleetState
+    previous_fleet_states: List[ArchivedFleetState]
+    known_nodes: Optional[List[RemoteUrsulaStatus]]
+
+    def to_json(self) -> Dict[str, Any]:
+        if self.known_nodes is None:
```

Comment:
Do you anticipate that `self.known_nodes` will be something other than `None` or an instance?

```suggestion
    if not self.known_nodes:
```

If it's a constant sorrow constant it's stronger to check identity against constant explicitly.

Reply:
It can potentially be an empty list (if known node info was requested, but `FleetSensor` doesn't know any nodes). So I think the check for `None` is justified here.